pkg/template/controller/templateinstance_controller.go

```diff
@@ -429,109 +410,101 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templateapi.T
+		if mappingErr != nil {
+			allErrors = append(allErrors, mappingErr)
+			continue
 		}
-		createObj, createErr := helper.Create(namespace, false, obj)
+
+		namespace := templateInstance.Namespace
+		if len(currObj.GetNamespace()) > 0 {
+			namespace = currObj.GetNamespace()
+		}
+		if namespace == "" {
+			allErrors = append(allErrors, errors.New("namespace was empty"))
+			continue
+		}
+
+		createObj, createErr := c.dynamicClient.Resource(restMapping.Resource).Namespace(namespace).Create(currObj)
 		if kerrors.IsAlreadyExists(createErr) {
-			createObj, createErr = obj, nil
-			obj, err := helper.Get(namespace, info.Name, false)
-			if err != nil {
-				return nil, err
+			freshGottenObj, getErr := c.dynamicClient.Resource(restMapping.Resource).Namespace(namespace).Get(currObj.GetName(), metav1.GetOptions{})
+			if getErr != nil {
+				allErrors = append(allErrors, getErr)
+				continue
 			}
 
-			meta, err := meta.Accessor(obj)
-			if err != nil {
-				return nil, err
-			}
-			labels := meta.GetLabels()
-			// no labels, so this isn't our object.
-			if labels == nil {
-				return createObj, createErr
-			}
-			owner, ok := labels[templateapi.TemplateInstanceOwner]
+			owner, ok := freshGottenObj.GetLabels()[templateapi.TemplateInstanceOwner]
 			// if the labels match, it's already our object so pretend we created
 			// it successfully.
 			if ok && owner == string(templateInstance.UID) {
-				createObj, createErr = obj, nil
+				createObj, createErr = freshGottenObj, nil
+			} else {
+				allErrors = append(allErrors, createErr)
+				continue
 			}
 		}
-
 		if createErr != nil {
-			return createObj, createErr
-		}
-
-		meta, err := meta.Accessor(createObj)
-		if err != nil {
-			return nil, err
+			allErrors = append(allErrors, mappingErr)
+			continue
 		}
 
 		templateInstance.Status.Objects = append(templateInstance.Status.Objects,
 			templateapi.TemplateInstanceObject{
 				Ref: kapi.ObjectReference{
-					Kind:       info.Mapping.GroupVersionKind.Kind,
+					Kind:       restMapping.GroupVersionKind.Kind,
 					Namespace:  namespace,
-					Name:       info.Name,
-					UID:        meta.GetUID(),
-					APIVersion: info.Mapping.GroupVersionKind.GroupVersion().String(),
+					Name:       createObj.GetName(),
+					UID:        createObj.GetUID(),
+					APIVersion: restMapping.GroupVersionKind.GroupVersion().String(),
 				},
 			},
 		)
-
-		return createObj, nil
 	}
 
-	// Second, create the objects, being tolerant if they already exist and are
-	// labelled as having previously been created by us.
-	glog.V(4).Infof("TemplateInstance controller: creating objects for %s/%s", templateInstance.Namespace, templateInstance.Name)
-
-	templateInstance.Status.Objects = nil
-
-	errs = bulk.Run(&kapi.List{Items: template.Objects}, templateInstance.Namespace)
+	// TODO figure out why this work is still done even on failures.  This was the way the code was before.
```

Comment:
because failures don't mean we didn't create *anything* it just means we didn't create *something* so we still need to add the finalizer so we can clean up the things we did add.